File: 0001.TwoSum/sol.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int len = nums.length;
        Map<Integer, Integer> map = new HashMap<>();
        int ret[] = new int[2];
        for(int i = 0; i < len; ++i){
            map.put(nums[i], i);
        }
        for(int i = 0; i < len; ++i){
            int dst = target - nums[i];
            if(map.containsKey(dst) && map.get(dst) != i){
                ret[0] = i;
                ret[1] = map.get(dst);
                return ret;
            }
        }
        return ret;
    }
}

File: 0001.TwoSum/sol.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int len = nums.Length;
        Dictionary<int, int> dic = new Dictionary<int, int>();
        for(int i = 0; i < len; ++i){
            if(dic.ContainsKey(nums[i])){
                dic[nums[i]] = i;
            }else{
                dic.Add(nums[i], i);
            }
        }
        int[] ret = new int[2];
        for(int i = 0; i < len; ++i){
            int dst = target - nums[i];
            if(dic.ContainsKey(dst)){
                // not itself
                if(dic[dst] != i){
                    ret[0] = i;
                    ret[1] = dic[dst];
                    return ret;
                }
            }
        }
        return ret;
    }
}